### PR TITLE
Add indeterminate state to checkbox

### PR DESCRIPTION
**Pull request by YgorSouza**

### CODE REVIEW

**Comment by YgorSouza** [comment]
[Screencast from 2023-11-21 21-38-32.webm](https://github.com/emilk/egui/assets/43298013/eb5ef1ac-d3f4-4f6d-a6a1-ea9d99eb0250)

**Comment by DataTriny** [comment]
Hello,

In case of an indeterminate checked state, `Checked::Mixed` must be passed to [`NodeBuilder::set_checked`](https://docs.rs/accesskit/latest/accesskit/struct.NodeBuilder.html#method.set_checked) when an accessible node is built [here](https://github.com/emilk/egui/blob/9253cafedda22a719ccfc7f898f56e8da631b3b0/crates/egui/src/response.rs#L618).

**Comment by DataTriny** [comment]
From an accessibility standpoint this is now OK, the state of the check box in the demo app is correctly picked up by ATs. Thanks!

File: crates/egui/src/widgets/button.rs [hunk 1]
```diff
@@ -355,24 +355,40 @@ impl Widget for Button<'_> {
...
             text: text.into(),
+            indeterminate: false,
         }
     }
 
     pub fn without_text(checked: &'a mut bool) -> Self {
         Self::new(checked, WidgetText::default())
     }
+
+    /// Display an indeterminate state (neither checked not unchecked)
```
Comment (emilk) [inline crates/egui/src/widgets/button.rs:374]:
```suggestion
    /// Display an indeterminate state (neither checked nor unchecked)
```

File: crates/egui/src/widgets/button.rs [hunk 2]
```diff
@@ -419,7 +439,16 @@ impl<'a> Widget for Checkbox<'a> {
...
             ));
 
-            if *checked {
+            if indeterminate {
+                // Horizontal line:
+                ui.painter().add(Shape::line(
+                    vec![
+                        pos2(small_icon_rect.left(), small_icon_rect.center().y),
+                        pos2(small_icon_rect.right(), small_icon_rect.center().y),
+                    ],
```
Comment (emilk) [inline crates/egui/src/widgets/button.rs:448]: nit: use `Shape::hline` for shorter code

**Review by emilk — APPROVED** [review]
Nice!

**Pull request merged**